libraries/standard/mqtt/cbmc/proofs/MQTT_Init/MQTT_Init_harness.c

```diff
@@ -25,16 +25,17 @@
  */
 
 #include "mqtt.h"
+#include "mqtt_cbmc_state.h"
 
 void harness()
 {
-    MQTTContext_t context;
-    MQTTTransportInterface_t transportInterface;
-    MQTTApplicationCallbacks_t callbacks;
-    MQTTFixedBuffer_t networkBuffer;
+    MQTTContext_t * pContext = mallocCanFail( sizeof( MQTTContext_t ) );
+    MQTTTransportInterface_t * pTransportInterface = mallocCanFail( sizeof( MQTTContext_t ) );
+    MQTTApplicationCallbacks_t * pCallbacks = mallocCanFail( sizeof( MQTTApplicationCallbacks_t ) );
+    MQTTFixedBuffer_t * pNetworkBuffer = mallocCanFail( sizeof( MQTTFixedBuffer_t ) );
 
-    MQTT_Init( nondet_bool() ? NULL : &context,
-               nondet_bool() ? NULL : &transportInterface,
-               nondet_bool() ? NULL : &callbacks,
-               nondet_bool() ? NULL : &networkBuffer );
+    MQTT_Init( pContext,
+               pTransportInterface,
+               pCallbacks,
+               pNetworkBuffer );
```

Comment:
You know, you could just use the mallocCanFail invocations in the MQTT_Init invocation and forget all about the type declarations above.

Reply:
yes, but for consistency I will just leave it.